beacon-chain/core/altair/attestation.go

```diff
@@ -295,8 +296,15 @@ func AttestationParticipationFlagIndices(beaconState state.BeaconState, data *et
 		participatedFlags[sourceFlagIndex] = true
 	}
 	matchedSrcTgt := matchedSrc && matchedTgt
-	if matchedSrcTgt && delay <= slotsPerEpoch {
-		participatedFlags[targetFlagIndex] = true
+	if beaconState.Version() >= version.Deneb {
+		// EIP-7045: Starting in Deneb, include attestations with a correct vote without considering a delay.
+		if matchedSrcTgt {
+			participatedFlags[targetFlagIndex] = true
+		}
+	} else {
+		if matchedSrcTgt && delay <= slotsPerEpoch {
```

Comment:
Do we really need this change? before Deneb no attestation should pass validation without having `delay <= slotsPerEpoch`.

Reply:
I am a bit torn. I see arguments for both.
1.) If the input attestations were validated, it's not-needed. The code is much cleaner
2.) If the input attestations were not validated then we need the filter here